overlord/snapstate/handlers.go

```diff
@@ -1942,6 +1942,115 @@ func (m *SnapManager) doPreferAliases(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// nearlyReady returns whether all lanes (other than me) are Ready.
+func nearlyReady(me string, change *state.Change) bool {
+	for _, task := range change.Tasks() {
+		if me == task.ID() {
+			// ignore self
+			continue
+		}
+		if !task.Status().Ready() {
+			return false
+		}
+	}
+	return true
+}
+
+// laneSnapSetups returns the instance names from the first SnapSetup in each
+// non-zero lane that is Done
+func laneSnaps(change *state.Change) []string {
+	// change.Tasks() preserves the order tasks were added, otherwise it all falls apart
+	tasks := change.Tasks()
+	sadLanes := map[int]bool{}
```

Comment:
not sure how worth its, but there should be a way to go over the tasks only once using some map

map[int]string  mapping   lane (!= 0) ==>  snapName|"<sad>"

once an entry is set to "<sad>" it stays that way,  once an entry is non empty no need to find / set the snapName in it